services/rds/src/main/java/org/finra/gatekeeper/services/db/connections/LambdaConnection.java

```diff
@@ -177,18 +141,50 @@ private String getGkUserName(String user, RoleType role){
     }
 
     public Map<RoleType, List<String>> getAvailableTables(RdsQuery rdsQuery) throws SQLException{
-        LambdaDTO lambdaDTO = new LambdaDTO().withDbEngine(rdsQuery.getDbEngine()).withLambdaQuery(new LambdaQuery(rdsQuery)).withGatekeeperPassword(gkUserCredentialsProvider.getGatekeeperSecret(rdsQuery));
+        return invokeHelper(new LambdaQuery(rdsQuery), "getAvailableSchemas", "POST", Map.class);
+    }
+
+    private <T> T invokeHelper(LambdaQuery lambdaQuery, String uri, String method, Class<T> clazz){
+        LambdaDTO lambdaDTO = new LambdaDTO()
+                .withDbEngine(lambdaQuery.getDbEngine())
+                .withLambdaQuery(lambdaQuery);
         try {
             String body = OBJECT_MAPPER.writeValueAsString(lambdaDTO);
-            String lambdaJson = invokeLambda("getAvailableSchemas", "POST", body).get("body").toString();
-            Map lambdaResult = OBJECT_MAPPER.readValue(lambdaJson, new TypeReference<Map<RoleType, List<String>>>(){});
-            return lambdaResult;
-        } catch (JsonProcessingException e) {
+            String lambdaJson = invokeLambda(uri, method, body).get("body").toString();
+            T lambdaResult;
+            switch (uri){
```

Comment:
You should be able to straight up use clazz instead of needing this switch statement to figure out the class object to use.